packages/http/src/utils/__tests__/parseResponse.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { parseResponse, parseResponseBody, parseResponseHeaders } from '../parseResponse';
+import { assertLeft, assertRight } from '../../__tests__/utils';
+import { Headers } from 'node-fetch';
+
+describe('parseResponseBody()', () => {
+  describe('body is json', () => {
+    describe('body is parseable', () => {
+      it('returns parsed body', async () => {
+        const response = {
+          headers: new Map([['content-type', 'application/json']]),
+          json: jest.fn().mockResolvedValue({ test: 'test' }),
+          text: jest.fn(),
+        } as any;
+
+        assertRight(
+          await parseResponseBody(response)(),
+          body => expect(body).toEqual({ test: 'test' }),
+        );
+
+        expect(response.text).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('body is not parseable', () => {
+      it('returns error', async () => {
+        const response = {
+          headers: new Map([['content-type', 'application/json']]),
+          json: jest.fn().mockRejectedValue(new Error('Big Bada Boom')),
+          text: jest.fn(),
+        } as any;
+
+        assertLeft(
+          await parseResponseBody(response)(),
+          error => expect(error.message).toEqual('Big Bada Boom'),
+        );
```

Comment:
This is probably my fault for the bad naming but I think we should create separate helpers for `TaskEither` so we do not have to execute the promise right now and have this weird `await` everywhere and `any` as returned body